ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -449,20 +481,42 @@ def add_veth_if_to_docker(self, ext_if, int_if):
                 VMTopology.cmd("nsenter -t %s -n ip link set dev %s mtu %d" % (self.pid, t_int_if, self.fp_mtu))
             elif int_if in self.cntr_ifaces:
                 VMTopology.cmd("nsenter -t %s -n ip link set dev %s mtu %d" % (self.pid, int_if, self.fp_mtu))
+            if create_vlan_subintf:
+                if t_int_sub_if in self.host_ifaces:
+                    VMTopology.cmd("ip link set dev %s mtu %d" % (t_int_sub_if, self.fp_mtu))
+                elif t_int_sub_if in self.cntr_ifaces:
+                    VMTopology.cmd("nsenter -t %s -n ip link set dev %s mtu %d" % (self.pid, t_int_sub_if, self.fp_mtu))
+                elif int_sub_if in self.cntr_ifaces:
+                    VMTopology.cmd("nsenter -t %s -n ip link set dev %s mtu %d" % (self.pid, int_sub_if, self.fp_mtu))
 
         VMTopology.iface_up(ext_if)
 
         self.update()
 
         if t_int_if in self.host_ifaces and t_int_if not in self.cntr_ifaces and int_if not in self.cntr_ifaces:
             VMTopology.cmd("ip link set netns %s dev %s" % (self.pid, t_int_if))
+        if (
+            create_vlan_subintf and
+            t_int_sub_if in self.host_ifaces and
+            t_int_sub_if not in self.cntr_ifaces and
+            int_sub_if not in self.cntr_ifaces
+        ):
+            VMTopology.cmd("ip link set netns %s dev %s" % (self.pid, t_int_sub_if))
 
         self.update()
 
         if t_int_if in self.cntr_ifaces and int_if not in self.cntr_ifaces:
             VMTopology.cmd("nsenter -t %s -n ip link set dev %s name %s" % (self.pid, t_int_if, int_if))
+        if (
+            create_vlan_subintf and
+            t_int_sub_if in self.cntr_ifaces and
+            int_sub_if not in self.cntr_ifaces
+        ):
+            VMTopology.cmd("nsenter -t %s -n ip link set dev %s name %s" % (self.pid, t_int_sub_if, int_sub_if))
 
         VMTopology.iface_up(int_if, self.pid)
+        if create_vlan_subintf:
+            VMTopology.iface_up(int_sub_if, self.pid)
```

Comment:
What about the ovs bindings? Don't we need to update that as well to use the sub interface?

Reply:
Do you mean the sub interfaces in the VM(`eos`)?

Reply:
Ignore. ptf_index is used in bind_ovs_ports()